### DISCUSSION TITLE
patch_match compiling failed

### DISCUSSION THREAD

**Discussion by Nezara** [lkwq007/stablediffusion-infinity#54] [Q&A]
I have installed opencv via `pip install opencv-python` but seem to be encountering a patch_match issue. Not sure whats causing it. Cheers.

(sd-inf) C:\Users\wstob\stablediffusion-infinity>python app.py
patch_match compiling failed
Traceback (most recent call last):
  File "C:\Users\wstob\stablediffusion-infinity\app.py", line 221, in <module>
    frame = gr.HTML(test(2), visible=False)
  File "C:\Users\wstob\stablediffusion-infinity\app.py", line 31, in test
    x = load_html()
  File "C:\Users\wstob\stablediffusion-infinity\app.py", line 23, in load_html
    canvaspy = f.read()
  File "C:\Users\wstob\anaconda3\envs\sd-inf\lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x8f in position 347: character maps to <undefined>

**Comment by lkwq007** [comment]
Seems to be encoding issues. I have updated the `app.py`. 

By the way, "patch_match compiling failed" is expected on windows

**Comment by badcode6** [comment]
installing opencv python will not help entirely with patch match because patch match is built with opencv C++/C library

**Comment by badcode6** [comment]
you also have to compile it and load it manually because the scripts are written for linux

**Comment by TemporalLabsLLC-SOL** [comment]
> you also have to compile it and load it manually because the scripts are written for linux

Have a link?

**Comment by badcode6** [comment]
> > you also have to compile it and load it manually because the scripts are written for linux
> 
> Have a link?

the modifications that need to be made to the C code are https://github.com/badcode6/PyPatchMatchWin/tree/master/csrc here. you need to build this with Visual Studio, under a DLL project. you do need the entire opencv library as stated in #27 . https://docs.opencv.org/4.x/dd/d6e/tutorial_windows_visual_studio_opencv.html this link should help you with that

**Comment by lkwq007** [comment]
> > > you also have to compile it and load it manually because the scripts are written for linux
> > 
> > 
> > Have a link?
> 
> the modifications that need to be made to the C code are https://github.com/badcode6/PyPatchMatchWin/tree/master/csrc here. you need to build this with Visual Studio, under a DLL project. you do need the entire opencv library as stated in #27 . https://docs.opencv.org/4.x/dd/d6e/tutorial_windows_visual_studio_opencv.html this link should help you with that

Actually I have tried compiling the dll library in last week. 
[libpatchmatch.zip](https://github.com/lkwq007/stablediffusion-infinity/files/9690584/libpatchmatch.zip)

Not sure whether this compiled dll works or not. On my windows pc, I have to load the `libpatchmatch.dll` before importing `numpy` and `PIL`. Otherwise it would result in `FileNotFoundError`.

**Comment by JamesIV4** [comment]
@lkwq007 Where should I extract these?

**Comment by lkwq007** [comment]
> @lkwq007 Where should I extract these?

@JamesIV4 
The latest version of this project can automatically download and load these DLL libraries. Please try the latest version. (Note that `git pull` might not work correctly due to a git submodule change. You may need to re-clone the repo)
